lib/epochs.js

```diff
@@ -0,0 +1,118 @@
+const { promisify: p } = require('util')
+const { fromMessageSigil, fromFeedSigil } = require('ssb-uri2')
+
+const Reduce = require('@tangle/reduce')
+const Strategy = require('@tangle/strategy')
+const OverwriteFields = require('@tangle/overwrite-fields')
+
+const getTangleUpdates = require('./tangles/get-tangle-updates')
+
+const msgPattern = [
+  '^',
+  'ssb:message',
+  '/',
+  '[a-zA-Z0-9-]+', // message "format"
+  '/',
+  '[a-zA-Z0-9-_]{43}=', // message "key"
+  '$'
+].join('')
+
+const feedPattern = [
+  '^',
+  'ssb:feed',
+  '/',
+  '[a-zA-Z0-9-_]+', // feed "format"
+  '/',
+  '[a-zA-Z0-9-_]{43}=', // feed "key"
+  '$'
+].join('')
+
+const strategy = OverwriteFields({
+  keyPattern: msgPattern,
+  valueSchema: {
+    type: 'object',
+    properties: ['author', 'epochKey'],
+    required: ['author', 'epochKey'],
+    author: {
+      type: 'string',
+      pattern: feedPattern
+    },
+    epochKey: {
+      type: 'string',
+      pattern: '[a-zA-Z0-9/\\+]{43}=' // 32 bytes in base64
+    }
+  }
+})
+strategy.mapToPure = T => T || strategy.identity() // function @tangle/reduce needs
```

Comment:
this is sort of the black box magic that i mentioned. i have no idea what this means. maybe that's ok though

Reply:
yeah ... it's because I'm using low level tools.
I will perhaps leave a longer comment about what `@tangle/reduce` is bringing to the party